processmanager/processinfo.go

```diff
@@ -88,11 +89,25 @@ func (pm *ProcessManager) updatePidInformation(pid libpf.PID, m *Mapping) (bool,
 		if name, err := os.ReadFile(fmt.Sprintf("/prod/%d/comm", pid)); err == nil {
 			processName = string(name)
 		}
+
+		envVarMap := make(map[string]string)
+		if envVars, err := os.ReadFile(fmt.Sprintf("/proc/%d/environ", pid)); err == nil {
+			splittedVars := strings.Split(string(envVars), "\000")
+			for _, envVar := range splittedVars {
+				keyValuePair := strings.SplitN(envVar, "=", 2)
+
+				if _, ok := pm.includeEnvVars[keyValuePair[0]]; ok {
```

Comment:
The last value in `/proc/PID/environ` is `00`. In hex it looks something like this:
```
47  44 4D 53 45  53 53 49 4F 4E 3D 69 33  00 44 49 53  50 4C 41 59  3D 3A 31 00
```
As `00` is not only the dividing element but also the limiting element here, when accessing `keyValuePair[0]` is panics. One option could be, to strip the final `00` before calling `strings.Split()`

Reply:
Suggestion: just check the length of `keyValuePair` here to be on the safe side.

Reply:
I am now validating the length of the keyValuePair (has to be = 2 or else entry is ignored)